lib/fluent/supervisor.rb

```diff
@@ -190,6 +190,12 @@ def install_supervisor_signal_handlers
         $log.debug 'fluentd supervisor process got SIGUSR2'
         supervisor_sigusr2_handler
       end
+
+      term_handler = trap :TERM do
+        # After SIGTERM, disable the sigdump file by ignoring SIGCONT.
+        trap(:CONT, nil)
+        term_handler.call
```

Comment:
This `term_handler.call()` will call `Server.stop()` in ServerEngine.
I think this is a nice way.

* https://github.com/treasure-data/serverengine/blob/v2.3.1/lib/serverengine/server.rb#L103